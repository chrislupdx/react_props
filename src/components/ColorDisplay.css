.ColorDisplay {
    width: 50mm;
    height: 50mm;
  }