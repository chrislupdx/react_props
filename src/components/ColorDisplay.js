import React from 'react';
import propTypes from 'prop-types';
import styles from './ColorDisplay.css';

function ColorDisplay({ color }) {
  return (
    <div ClassName={styles.ColorDisplay} style={{ backgroundColor: color }}></div>
  );
}

ColorDisplay.propTypes = {
  color: propTypes.string.isRequired
};

export default ColorDisplay;
