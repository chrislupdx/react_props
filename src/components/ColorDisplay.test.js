import React from 'react';
import { shallow } from 'enzyme';
import ColorDisplay from './ColorDisplay';

describe('ColorDisplay componenet', () => {
  it('renders a colordisplay', () => {
    const wrapper = shallow(<ColorDisplay color ="#FF0000" />);
    expect(wrapper).toMatchSnapshot();
  });
})
;
