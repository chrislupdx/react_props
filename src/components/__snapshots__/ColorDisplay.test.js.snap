// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorDisplay componenet renders a colordisplay 1`] = `
<div
  ClassName="ColorDisplay"
  style={
    Object {
      "backgroundColor": "#FF0000",
    }
  }
/>
`;
